src/Relay.ts

```diff
@@ -0,0 +1,103 @@
+interface Message {
+  to?: string
+  message?:string
+  payload?:any
+  sender?:string
+}
+
+class Relay extends EventTarget {
+  id;
+  name;
+  connections = new Map();
+
+  constructor(name:string) {
+    super();
+    this.name = name;
+    this.broadcast = this.broadcast.bind(this);
+    this.send = this.send.bind(this);
+    this.addConnection = this.addConnection.bind(this);
+    this.removeConnection = this.removeConnection.bind(this);
+    this.createEvent = this.createEvent.bind(this);
+  }
+
+  addConnection(name:string, fn:EventListener) {
+    function wrappedFn(event:CustomEvent) { 
+      return fn(event.detail); 
+    };
+
+    this.addEventListener(name, wrappedFn);
+    this.connections.set(name, wrappedFn);
+
+    return () => this.removeConnection(name);
+  }
+
+  removeConnection(name:string) {
+    const fn:EventListener = this.connections.get(name);
+    this.removeEventListener(name, fn);
+    this.connections.delete(name);
+  }
+
+  dispatch(message) {
+    this.dispatchEvent(message);
+  }
+
+  createEvent(message:string, detail:Message = {}) {
+    return new CustomEvent(message, { detail });
+  }
+
+  broadcast(message:Message, sender) {
```

Comment:
What's the type of `sender` here? I think you'll be glad you went to the trouble of specifying types for everything you can in this module, since it's so central to everything you're doing.